### PR TITLE
Warn when using C++14 in CUB and Thrust

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR adds a deprecation warning when using CUB or Thrust in C++14 mode.

### CODE REVIEW

File: cub/cub/cmake/cub-config.cmake [hunk 1]
```diff
@@ -103,6 +103,11 @@ if (CUB_IGNORE_DEPRECATED_CPP_11 OR
   target_compile_definitions(_CUB_CUB INTERFACE "CUB_IGNORE_DEPRECATED_CPP_11")
 endif()
 
+if (CUB_IGNORE_DEPRECATED_CPP_14 OR
+        THRUST_IGNORE_DEPRECATED_CPP_14)
```
Comment (alliepiper) [inline cub/cub/cmake/cub-config.cmake:107]: nitpick: indention

File: cub/cub/util_cpp_dialect.cuh [hunk 2]
```diff
@@ -118,7 +122,10 @@ CUB_COMPILER_DEPRECATION_SOFT(MSVC 2019(19.20 / 16.0 / 14.20), MSVC 2017);
 CUB_COMPILER_DEPRECATION(C++ 14);
 #    elif _CCCL_STD_VER == 2011 && !defined(CUB_IGNORE_DEPRECATED_CPP_11)
 // =C++11. Soft upgrade message:
-CUB_COMPILER_DEPRECATION_SOFT(C++ 14, C++ 11);
+CUB_COMPILER_DEPRECATION_SOFT(C++ 17, C++ 11);
+#    elif _CCCL_STD_VER == 2014 && !defined(CUB_IGNORE_DEPRECATED_CPP_14)
+// =C++14. Soft upgrade message:
+CUB_COMPILER_DEPRECATION_SOFT(C++ 17, C++ 11);
```
Comment (alliepiper) [inline cub/cub/util_cpp_dialect.cuh:128]:
```suggestion
CUB_COMPILER_DEPRECATION_SOFT(C++ 17, C++ 14);
```

**Review by alliepiper — APPROVED** [review]
Found a couple other places that need updating:

- https://github.com/NVIDIA/cccl/blob/24ed47daab316859c84652b19f563b2f355f0241/thrust/thrust/cmake/thrust-config.cmake#L56
- https://github.com/NVIDIA/cccl/blob/24ed47daab316859c84652b19f563b2f355f0241/cub/cmake/CubBuildTargetList.cmake#L180
- https://github.com/NVIDIA/cccl/blob/24ed47daab316859c84652b19f563b2f355f0241/thrust/cmake/ThrustBuildTargetList.cmake#L303

Otherwise this LGTM

**Comment by bernhardmgruber** [comment]
@alliepiper thanks for the great feedback! I applied your changes.

**Pull request merged**